slp_abacus/slp_abacus/parse/abacus_parser.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+from otm.otm.entity.otm import OTM
+from otm.otm.otm_builder import OTMBuilder
+from slp_base import OTMBuildingError
+from slp_base.slp_base.provider_parser import ProviderParser
+from slp_base.slp_base.provider_type import DiagramType
+from slp_abacus.slp_abacus.load.abacus_mapping_file_loader import AbacusMapping
+from slp_abacus.slp_abacus.objects.diagram_objects import Diagram
+from slp_abacus.slp_abacus.parse.diagram_mapper import DiagramMapper
+from slp_abacus.slp_abacus.parse.transformers.default_trustzone_transformer import DefaultTrustZoneTransformer
+from slp_abacus.slp_abacus.parse.transformers.parent_calculator_transformer import ParentCalculatorTransformer
+
+logger = logging.getLogger(__name__)
+
+
+class AbacusParser(ProviderParser):
+    """
+    Parser to build an OTM from Abacus
+    """
+
+    def __init__(self, project_id: str, project_name: str, diagram: Diagram, mapping: AbacusMapping):
+        self.project_id = project_id
+        self.project_name = project_name
+        self.diagram = diagram
+        self.mapping = mapping
+
+    def build_otm(self) -> OTM:
+        try:
+            DiagramMapper(self.diagram, self.mapping).map()
+
+            ParentCalculatorTransformer(self.diagram).transform()
```

Comment:
We are not calculating parents at this point, so it is dead code. Even if in the future we could support that, we usually prefer to not add useless code unless special cases.